From 24aab57de52a7c599ab273322103b497b1ff24db Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Mon, 19 Jun 2023 09:44:28 +0200
Subject: [PATCH] ARM: dts: stm32: fix dmas entres for CRYP driver

MDMA on take 5 32bits word in is DMA cell.
Remove one 0x00 to obtain a valid configuration.

Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Change-Id: If3804e5a513b8ff38107471bf798829637b9c3da
---
 arch/arm/boot/dts/st/stm32mp13xc.dtsi | 4 ++--
 arch/arm/boot/dts/st/stm32mp13xf.dtsi | 4 ++--
 arch/arm/boot/dts/st/stm32mp15xc.dtsi | 3 +++
 arch/arm/boot/dts/st/stm32mp15xf.dtsi | 3 +++
 4 files changed, 10 insertions(+), 4 deletions(-)

diff --git a/arch/arm/boot/dts/st/stm32mp13xc.dtsi b/arch/arm/boot/dts/st/stm32mp13xc.dtsi
index b49b9b9c37f5..5ed8fdd40cd3 100644
--- a/arch/arm/boot/dts/st/stm32mp13xc.dtsi
+++ b/arch/arm/boot/dts/st/stm32mp13xc.dtsi
@@ -11,8 +11,8 @@ cryp: crypto@54002000 {
 		interrupts = <GIC_SPI 80 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&rcc CRYP1>;
 		resets = <&rcc CRYP1_R>;
-		dmas = <&mdma 28 0x0 0x400202 0x0 0x0 0x0>,
-		       <&mdma 29 0x3 0x400808 0x0 0x0 0x0>;
+		dmas = <&mdma 28 0x0 0x400202 0x0 0x0>,
+		       <&mdma 29 0x3 0x400808 0x0 0x0>;
 		dma-names = "in", "out";
 		access-controllers = <&etzpc 42>;
 		status = "disabled";
diff --git a/arch/arm/boot/dts/st/stm32mp13xf.dtsi b/arch/arm/boot/dts/st/stm32mp13xf.dtsi
index b49b9b9c37f5..5ed8fdd40cd3 100644
--- a/arch/arm/boot/dts/st/stm32mp13xf.dtsi
+++ b/arch/arm/boot/dts/st/stm32mp13xf.dtsi
@@ -11,8 +11,8 @@ cryp: crypto@54002000 {
 		interrupts = <GIC_SPI 80 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&rcc CRYP1>;
 		resets = <&rcc CRYP1_R>;
-		dmas = <&mdma 28 0x0 0x400202 0x0 0x0 0x0>,
-		       <&mdma 29 0x3 0x400808 0x0 0x0 0x0>;
+		dmas = <&mdma 28 0x0 0x400202 0x0 0x0>,
+		       <&mdma 29 0x3 0x400808 0x0 0x0>;
 		dma-names = "in", "out";
 		access-controllers = <&etzpc 42>;
 		status = "disabled";
diff --git a/arch/arm/boot/dts/st/stm32mp15xc.dtsi b/arch/arm/boot/dts/st/stm32mp15xc.dtsi
index f20046722753..55ca2814f89e 100644
--- a/arch/arm/boot/dts/st/stm32mp15xc.dtsi
+++ b/arch/arm/boot/dts/st/stm32mp15xc.dtsi
@@ -13,6 +13,9 @@ cryp1: cryp@54001000 {
 		interrupts = <GIC_SPI 79 IRQ_TYPE_LEVEL_HIGH>;
 		clocks = <&rcc CRYP1>;
 		resets = <&rcc CRYP1_R>;
+		dmas = <&mdma1 29 0x0 0x400202 0x0 0x0>,
+		       <&mdma1 30 0x3 0x400808 0x0 0x0>;
+		dma-names = "in", "out";
 		access-controllers = <&etzpc 9>;
 		status = "disabled";
 	};
diff --git a/arch/arm/boot/dts/st/stm32mp15xf.dtsi b/arch/arm/boot/dts/st/stm32mp15xf.dtsi
index 62a5fd77b1fc..8134f7ec6282 100644
--- a/arch/arm/boot/dts/st/stm32mp15xf.dtsi
+++ b/arch/arm/boot/dts/st/stm32mp15xf.dtsi
@@ -14,6 +14,9 @@ cryp1: cryp@54001000 {
 		clocks = <&rcc CRYP1>;
 		resets = <&rcc CRYP1_R>;
 		access-controllers = <&etzpc 9>;
+		dmas = <&mdma1 29 0x0 0x400202 0x0 0x0>,
+		       <&mdma1 30 0x3 0x400808 0x0 0x0>;
+		dma-names = "in", "out";
 		status = "disabled";
 	};
 };
-- 
2.39.5

